pkg/maintainer/wallet/chain_test.go

```diff
@@ -0,0 +1,175 @@
+package wallet
+
+import (
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+	"github.com/keep-network/keep-core/pkg/coordinator"
+	"github.com/keep-network/keep-core/pkg/tbtc"
+)
+
+type walletLock struct {
+	lockExpiration time.Time
+	walletAction   tbtc.WalletActionType
+}
+
+type localChain struct {
+	mutex sync.Mutex
+
+	walletLocks map[[20]byte]*walletLock
+}
+
+func newLocalChain() *localChain {
```

Comment:
I realized we have a neat pattern we agreed on early in [`pkg/sortition`](https://github.com/keep-network/keep-core/tree/main/pkg/sortition) where we define the local chain implementation in `pkg/internal/local`. This way, `_test.go` is only for actual tests and we can unit-test the local chain implementation as well. I know we were not very consistent in using this pattern, but this refactor feels like a good opportunity to align at least this package.

cc @lukasz-zimnoch @tomaszslabon

Reply:
This pattern can cause import cycles. Suppose we implement `tbtc.Chain` in `pkg/tbtc/internal/local.go`, then the structure declared in this internal package must refer to `tbtc` chain types used by `tbtc.Chain`. On the other hand, the unit tests living in `tbtc` package must create an instance of `local.Chain`. Here is why this pattern is used only in `pkg/sorititon` where `sorition.Chain` does not declare any chain types.

Reply:
... and that's why I think we should have [the types declared in a separate package](https://github.com/keep-network/keep-core/pull/3650#discussion_r1242336911) 😁

Reply:
> I know we were not very consistent in using this pattern, but this refactor feels like a good opportunity to align at least this package.

IMHO if agree on a specific pattern we should refactor and apply it to the whole keep-core project. Currently, we have multiple patterns that we use randomly in different packages. When we create a new package we're not sure which pattern to follow, and we add even more inconsistencies.

Reply:
Since it doesn't look like we have a consistent idea of the approach we want to follow, and it seems out-of-scope of the current PR and https://github.com/keep-network/keep-core/issues/3632, I propose to handle a refactor of all local chain implementations in a separate PR (along with other changes like one mutex for the local chain you proposed recently).